packages/block-library/src/term-template/edit.js

```diff
@@ -0,0 +1,340 @@
+		if ( ! terms || parent !== 0 ) {
+			return terms;
+		}
+		return terms.filter( ( term ) => ! term.parent );
+	}, [ terms, parent ] );
+
+	const { blocks } = useSelect(
+		( select ) => ( {
+			blocks: select( blockEditorStore ).getBlocks( clientId ),
+		} ),
+		[ clientId ]
+	);
+
+	const blockContexts = useMemo(
+		() =>
+			filteredTerms?.map( ( term ) => ( {
+				taxonomy,
+				termId: term.id,
+				classList: `term-${ term.id }`,
+				termData: term,
+			} ) ),
+		[ filteredTerms, taxonomy ]
+	);
+
+	const blockProps = useBlockProps();
+
+	if ( isResolving ) {
+		return (
+			<div { ...blockProps }>
+				<ul>
+					<li className="wp-block-term term-loading">
+						<div className="term-loading-placeholder" />
+					</li>
+					<li className="wp-block-term term-loading">
+						<div className="term-loading-placeholder" />
+					</li>
+					<li className="wp-block-term term-loading">
+						<div className="term-loading-placeholder" />
+					</li>
+				</ul>
+			</div>
+		);
+	}
+
+	if ( ! filteredTerms?.length ) {
+		return <p { ...blockProps }> { __( 'No terms found.' ) }</p>;
+	}
+
+	const renderTerm = ( term ) => {
+		const blockContext = {
+			taxonomy,
+			termId: term.id,
+			classList: `term-${ term.id }`,
+			termData: term,
+		};
+
+		return (
+			<BlockContextProvider key={ term.id } value={ blockContext }>
+				{ isActiveTerm(
+					term.id,
+					activeBlockContextId,
+					blockContexts
+				) ? (
+					<TermTemplateInnerBlocks
+						classList={ blockContext.classList }
+					/>
+				) : null }
+				<MemoizedTermTemplateBlockPreview
+					blocks={ blocks }
+					blockContextId={ term.id }
+					classList={ blockContext.classList }
+					setActiveBlockContextId={ setActiveBlockContextId }
+					isHidden={ isActiveTerm(
+						term.id,
+						activeBlockContextId,
+						blockContexts
+					) }
+				/>
+			</BlockContextProvider>
+		);
+	};
+
+	const renderTerms = () => {
+		if ( hierarchical ) {
+			const termsTree = buildTermsTree( filteredTerms );
+			return termsTree.map( ( termNode ) =>
+				renderTermNode( termNode, renderTerm )
+			);
+		}
+
+		return blockContexts.map( ( blockContext ) => (
+			<BlockContextProvider
+				key={ blockContext.termId }
+				value={ blockContext }
+			>
+				{ isActiveTerm(
+					blockContext.termId,
+					activeBlockContextId,
+					blockContexts
+				) ? (
+					<TermTemplateInnerBlocks
+						classList={ blockContext.classList }
+					/>
+				) : null }
+				<MemoizedTermTemplateBlockPreview
+					blocks={ blocks }
+					blockContextId={ blockContext.termId }
+					classList={ blockContext.classList }
+					setActiveBlockContextId={ setActiveBlockContextId }
+					isHidden={ isActiveTerm(
+						blockContext.termId,
+						activeBlockContextId,
+						blockContexts
+					) }
+				/>
+			</BlockContextProvider>
+		) );
+	};
+
+	return (
+		<>
+			<BlockControls>
+				<ToolbarGroup />
+			</BlockControls>
+
+			<div { ...blockProps }>
+				<ul>{ renderTerms() }</ul>
+			</div>
```

Comment:
Feels like attaching block props to `ul` directly would make layouting easier.

```suggestion
				<ul { ...blockProps }>{ renderTerms() }</ul>
```

And if so, the same change could be applied in the loading skeleton as well.

Reply:
Yeah that looks better, thanks! Done in https://github.com/WordPress/gutenberg/pull/70720/commits/56e244d1e88a486fbfad98298daecd49339bbc2c.